protocols/identify/src/behaviour.rs

```diff
@@ -214,6 +263,57 @@ impl Behaviour {
             .cloned()
             .collect()
     }
+
+    fn emit_new_external_addr_candidate_event(
+        &mut self,
+        connection_id: ConnectionId,
+        observed: &Multiaddr,
+    ) {
+        if self
+            .outbound_connections_without_port_reuse
+            .contains(&connection_id)
+        {
+            // Apply address translation to the candidate address.
+            // For TCP without port-reuse, the observed address contains an ephemeral port which needs to be replaced by the port of a listen address.
+            let translated_addresses = {
+                let mut addrs: Vec<_> = self
+                    .listen_addresses
+                    .iter()
+                    .filter_map(|server| {
+                        if (is_tcp_addr(server) && is_tcp_addr(&observed))
+                            || (is_quic_addr(server, true) && is_quic_addr(&observed, true))
+                            || (is_quic_addr(server, false) && is_quic_addr(&observed, false))
+                        {
+                            address_translation(server, &observed)
+                        } else {
+                            None
+                        }
+                    })
+                    .collect();
+
+                // remove duplicates
+                addrs.sort_unstable();
+                addrs.dedup();
+                addrs
+            };
+
+            // If address translation yielded nothing, broadcast the original candidate address.
+            if translated_addresses.is_empty() {
+                self.events
+                    .push_back(ToSwarm::NewExternalAddrCandidate(observed.clone()));
+            } else {
+                for addr in translated_addresses {
+                    self.events
+                        .push_back(ToSwarm::NewExternalAddrCandidate(addr));
+                }
+            }
+        } else {
+            // outgoing connection dialed with port reuse
+            // incomming connection
+            self.events
+                .push_back(ToSwarm::NewExternalAddrCandidate(observed.clone()));
+        }
```

Comment:
```suggestion
        if !self
            .outbound_connections_without_port_reuse
            .contains(&connection_id)
        {
        	
            // outgoing connection dialed with port reuse
            // incomming connection
            self.events
                .push_back(ToSwarm::NewExternalAddrCandidate(observed.clone()));
            return;  
          }
          
          // Apply address translation to the candidate address.
          // For TCP without port-reuse, the observed address contains an ephemeral port which needs to be replaced by the port of a listen address.
          let translated_addresses = {
              let mut addrs: Vec<_> = self
                  .listen_addresses
                  .iter()
                  .filter_map(|server| {
                      if (is_tcp_addr(server) && is_tcp_addr(&observed))
                          || (is_quic_addr(server, true) && is_quic_addr(&observed, true))
                          || (is_quic_addr(server, false) && is_quic_addr(&observed, false))
                      {
                          address_translation(server, &observed)
                      } else {
                          None
                      }
                  })
                  .collect();

              // remove duplicates
              addrs.sort_unstable();
              addrs.dedup();
              addrs
          };

          // If address translation yielded nothing, broadcast the original candidate address.
          if translated_addresses.is_empty() {
              self.events
                  .push_back(ToSwarm::NewExternalAddrCandidate(observed.clone()));
          } else {
              for addr in translated_addresses {
                  self.events
                      .push_back(ToSwarm::NewExternalAddrCandidate(addr));
              }
          }
```

I'd prefer to avoid `else`.